src/main/java/uk/gov/hmcts/darts/audio/service/impl/ViqHeaderServiceImpl.java

```diff
@@ -23,8 +40,22 @@ public String generateAnnotation(Integer hearingId, String startTime, String end
         throw new NotImplementedException(VIQ_HEADER_SERVICE_NOT_SUPPORTED_MESSAGE);
     }
 
+    @SuppressWarnings({"PMD.AvoidFileStream"})
     @Override
-    public String generateReadme(ViqMetaData viqMetaData) {
-        throw new NotImplementedException(VIQ_HEADER_SERVICE_NOT_SUPPORTED_MESSAGE);
+    public String generateReadme(ViqMetaData viqMetaData, String fileLocation) {
+        File readmeFile = new File(fileLocation, README_TXT_FILENAME);
+        log.debug("Writing readme to {}", readmeFile.getAbsoluteFile());
+        try (BufferedWriter fileWriter = Files.newBufferedWriter(readmeFile.toPath());
+             PrintWriter printWriter = new PrintWriter(fileWriter);) {
+
+            printWriter.println(format(COURTHOUSE_README_LABEL + README_FORMAT, viqMetaData.getCourthouse()));
+            printWriter.println(format(RAISED_BY_README_LABEL + README_FORMAT, viqMetaData.getRaisedBy()));
+            printWriter.println(format(START_TIME_README_LABEL + README_FORMAT, viqMetaData.getStartTime()));
+            printWriter.println(format(END_TIME_README_LABEL + README_FORMAT, viqMetaData.getEndTime()));
+            printWriter.print(format(REQUEST_TYPE_README_LABEL + README_FORMAT, viqMetaData.getType()));
+        } catch (IOException exception) {
+            throw new DartsApiException(AudioError.FAILED_TO_PROCESS_AUDIO_REQUEST, exception);
```

Comment:
I'm not sure where this service will be called from at the moment, but it's probably worth checking @chrisbellingham-hmcts  is ok with you using his `AudioError` here. It might be that we are using the wrong exception here - should it be `DartsException` as I wouldn't expect this to be returned back to the user in an error problem response?

Reply:
Created a new error